mesos_master/datadog_checks/mesos_master/mesos_master.py

```diff
@@ -135,21 +135,32 @@ class MesosMaster(AgentCheck):
         'master/valid_status_updates': ('mesos.cluster.valid_status_updates', GAUGE),
     }
 
-    def __init__(self, name, init_config, agentConfig, instances=None):
-        AgentCheck.__init__(self, name, init_config, agentConfig, instances)
-        for instance in instances or []:
-            url = instance.get('url', '')
+    HTTP_CONFIG_REMAPPER = {'disable_ssl_validation': {'name': 'tls_verify', 'invert': True, 'default': False}}
+
+    def __init__(self, name, init_config, instances):
+        super(MesosMaster, self).__init__(name, init_config, instances)
+        if self.instance is not None:
+            url = self.instance.get('url', '')
             parsed_url = urlparse(url)
-            ssl_verify = not _is_affirmative(instance.get('disable_ssl_validation', False))
-            if not ssl_verify and parsed_url.scheme == 'https':
-                self.log.warning('Skipping SSL cert validation for {0} based on configuration.'.format(url))
 
-    def _get_json(self, url, timeout, verify=True, failure_expected=False, tags=None):
+            if not self.http.options['verify'] and parsed_url.scheme == 'https':
+                self.log.warning('Skipping TLS cert validation for %s based on configuration.' % url)
+            if not ('read_timeout' in self.instance or 'connect_timeout' in self.instance):
+                # `default_timeout` config option will be removed with Agent 5
+                self.http.options['timeout'] = (
+                    self.instance.get('timeout')
+                    or self.init_config.get('timeout')
+                    or self.init_config.get('default_timeout')
+                    or self.DEFAULT_TIMEOUT
+                )
```

Comment:
Sorry for reviewing a bit late. I think we need some tests for logics like this. I know it wasn't tested before, but we could maybe start now :)